From 90b5b48734b8738fac217df1b2846997f532a0ea Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 2 Jul 2015 14:55:31 +0200
Subject: [PATCH] dracut.sh: add devices with x-initrd.mount in /etc/fstab to
 host_devs

otherwise dracut might not even be able to mount those.

(cherry picked from commit 916559e073a49eedbc7b0c58b399d303fb8c2a05)
---
 dracut.sh | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/dracut.sh b/dracut.sh
index 6e56af50..0e600220 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -1061,6 +1061,20 @@ if [[ $hostonly ]]; then
             done < /etc/fstab
         done < /proc/swaps
     fi
+
+    # collect all "x-initrd.mount" entries from /etc/fstab
+    if [[ -f /etc/fstab ]]; then
+        while read _d _m _t _o _r || [ -n "$_d" ]; do
+            [[ "$_d" == \#* ]] && continue
+            [[ $_d ]] || continue
+            [[ "$_o" != *x-initrd.mount* ]] && continue
+            _dev=$(expand_persistent_dev "$_d")
+            _dev="$(readlink -f "$_dev")"
+            [[ -b $_dev ]] && push host_devs "$_dev"
+        done < /etc/fstab
+    fi
+
+
     # record all host modaliases
     declare -A host_modalias
     find  /sys/devices/ -name uevent -print > "$initdir/.modalias"
